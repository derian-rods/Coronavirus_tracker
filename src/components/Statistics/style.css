.app__cardsStats{
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-around;   
    width: 50vw;
    padding: 2em;
}
/* 
@media screen and (max-width: 1024px){
    .app__cardsStats{
        display: flex;
        flex-direction: row;
        flex-wrap: nowrap;
        justify-content: flex-start;
        padding: 0;
        width: 60vw;
    }
} */

/* 768px */
@media screen and (max-width: 1024px){
    .app__cardsStats{
        display: flex;
        flex-direction: row;
        justify-content: center;
        flex-wrap: nowrap;
        padding: 0;
        margin-left: 4em;
        width: 47em;
    }
}


@media screen and (max-width: 834px){
    .app__cardsStats{
        display: flex;
        flex-direction: row;
        justify-content: center;
        flex-wrap: nowrap;
        padding: 0;
        margin-left: -1.5em;
        width: 45em;
    }
}


/* 768px */
@media screen and (max-width: 768px){
    .app__cardsStats{
        display: flex;
        flex-direction: row;
        justify-content: center;
        flex-wrap: nowrap;
        padding: 0;
        margin-left: -3em;
        width: 45em;
    }
}

@media screen and (max-width: 414px){
    .app__cardsStats{
        display: flex;
        flex-direction: column-reverse;
        justify-content: center;
        padding: 0;
        margin-left: 2.5em;
        width: 10em;
    }
}

@media screen and (max-width: 411px){
    .app__cardsStats{
        display: flex;
        flex-direction: column-reverse;
        justify-content: center;
        margin-left: 3em;
        padding: 0;
    }
}
/* media 360px */
@media screen and (max-width: 360px){
    .app__cardsStats{
        display: flex;
        flex-direction: column-reverse;
        justify-content: center;
        margin-left: 3em;
        padding: 0;
    }
}
