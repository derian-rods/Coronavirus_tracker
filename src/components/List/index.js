import React from 'react'
import PropTypes from 'prop-types'
import './style.css'
export const List = ({countries}) => {
    return (
     <>
                <div className="app__list">
                <div className="list">
                    <h3 className="list__title">Cases By Country</h3>
                    <ul className="list__ul">
                    {
                            countries.map(country => (
                                <li className="list__li" key={country.country}>
                                   <div>
                                       <img className="list__flag" src={country.countryInfo.flag} alt={country.country} />
                                   <span>{country.country}</span>
                                   </div>
                                    <span>{country.cases}<i className="fas fa-virus list__cases"></i></span>
                                </li>
                            ))
                        }
                    </ul>
                </div>
            </div>
     </>
    )
}

    List.propTypes = {
        countries: PropTypes.array.isRequired,
    }