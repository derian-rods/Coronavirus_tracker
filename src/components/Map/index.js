import React, { useEffect, useState } from "react";
import { scaleQuantile } from "d3-scale";
import {Table} from './Table'
import {
  ComposableMap,
  Geographies,
  Geography,
  Sphere,
  ZoomableGroup,
  Graticule,
} from "react-simple-maps";
import PropTypes from 'prop-types'
import './style.css'

const geoUrl =
  "https://raw.githubusercontent.com/zcreativelabs/react-simple-maps/master/topojson-maps/world-110m.json";

  
  const MapChart = ({countries}) => {
    const [data, setData] = useState([]);
    
    const colorScale = scaleQuantile()
      .domain(data.map(d => d.cases))
      .range([   
        "#ffedea",
        "#ffcec5",
        "#ffad9f",
        "#ff8a75",
        "#ff5533",
        "#e2492d",
        "#be3d26",
        "#9a311f",
        "#782618",
    ]);
      
  useEffect(() => {
    
   const valueInfo = countries.map(country =>({
      name: country.country,
      cases: country.cases,
      iso3: country.countryInfo.iso3,
      deaths: country.deaths,
      recovered: country.recovered
    }));

    setData(valueInfo)
  
  },[countries]);  
  return (
    <div className="app__map">
        <div className="card__map">
          <h2 className="map__title">World Map</h2>
          <div className="map__table">
             <Table />
          </div>
        <ComposableMap
                projectionConfig={{
                    rotate: [-10, 0, 0],
                    scale: 160
                }}
                >
                <ZoomableGroup>
                <Sphere stroke="#E4E5E6" strokeWidth={0.5} />
                <Graticule stroke="#E4E5E6" strokeWidth={0.5} />
                {data.length > 0 && (
                    <Geographies geography={geoUrl}>
                    {({ geographies }) =>
                        geographies.map((geo) => {
                        const d = data.find((s) => s.iso3 === geo.properties.ISO_A3);
                        return (
                            <Geography
                            key={geo.rsmKey}
                            geography={geo}
                            fill={d ? colorScale(d.cases) : "#ffedea"}
                            />
                        );
                        })
                    }
                    </Geographies>
                )}
                </ZoomableGroup>
                </ComposableMap>
        </div>
    </div>
  );
};

MapChart.propTypes = {
  countries: PropTypes.array.isRequired,
}


export default MapChart;