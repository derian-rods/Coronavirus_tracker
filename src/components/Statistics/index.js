 import React from 'react'
import { Card } from '../Layouts/cards';
import PropTypes from 'prop-types'
import './style.css'
 
 export const CardStats = ({data}) => {
     return (
         <div className="app__cardsStats">
             <Card  title='Coronavirus cases' total={data.cases} increment={data.todayCases} icon={'fas fa-viruses'} icon2={'fas fa-angle-double-up'}  color={'redcolor'}/>
             <Card  title='Deaths' total={data.deaths} increment={data.todayDeaths} icon={'fas fa-skull-crossbones'} icon2={'fas fa-angle-double-up'}  color={'redcolor'} />
             <Card title='Recovered' total={data.recovered} increment={data.todayRecovered} icon={'fas fa-ambulance'} icon2={'fas fa-angle-double-up'}  color={'greencolor'}/>
         </div>
     )
 }
 
 CardStats.propTypes = {
     data: PropTypes.object.isRequired
 }