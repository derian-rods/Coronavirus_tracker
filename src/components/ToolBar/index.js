import React from 'react';
import PropTypes from 'prop-types'
import './style.css';

export const ToolBar = ({ countries, country, handleCountryChange}) => {

    return (
        <>
                <div className="navbar">
                <div className="navbar__brand">
                 <img className="brand__img" src='https://www.flaticon.com/svg/static/icons/svg/2785/2785819.svg' alt="corona virus"></img>
                 <span className="brand__text">Tracker</span>
                </div>
                <div className="navbar__card">
                     <select value={country} onChange={handleCountryChange}>
                         <option >World Wide</option>
                         {countries.map(c => (
                             <option key={c.country}>{c.country}</option>
                         ))}
                     </select>
                </div>
             </div>
        </>
    )
}

ToolBar.propTypes = {
    countries: PropTypes.array.isRequired,
    country: PropTypes.string.isRequired,
    hadleCountryChange: PropTypes.func
}