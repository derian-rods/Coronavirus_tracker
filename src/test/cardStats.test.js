import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom'
import { CardStats } from '../components/Statistics';

describe('Test for component CardStates', () => {
    test('should show <CardStats /> correctly ', () => {
        const data = {
            active: 232313,
            cases: 232012141,
            critical: 2332,
        }
        const wrapper = shallow (<CardStats data={data} />)
        expect(wrapper).toMatchSnapshot()

    });
})
