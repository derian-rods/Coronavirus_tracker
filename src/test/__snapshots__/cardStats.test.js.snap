// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for component CardStates should show <CardStats /> correctly  1`] = `
<div
  className="app__cardsStats"
>
  <Card
    color="redcolor"
    icon="fas fa-viruses"
    icon2="fas fa-angle-double-up"
    title="Coronavirus cases"
    total={232012141}
  />
  <Card
    color="redcolor"
    icon="fas fa-skull-crossbones"
    icon2="fas fa-angle-double-up"
    title="Deaths"
  />
  <Card
    color="greencolor"
    icon="fas fa-ambulance"
    icon2="fas fa-angle-double-up"
    title="Recovered"
  />
</div>
`;
